api/docs/reference/search_algorithm.md

```diff
@@ -0,0 +1,226 @@
+
+> **Note**: We also apply cluster-level configurations as part of our
+> Elasticsearch deployment. These are intentionally not covered here as they
+> primarily deal with cluster performance and are irrelevant to the way searches
+> are executed.
+
+## Text analysis/tokenization
+
+> **Note**: A general understanding of
+> [full-text search (Wikipedia)](https://en.wikipedia.org/wiki/Full-text_search)
+> and the concepts of
+> ["stemming" (Wikipedia)](https://en.wikipedia.org/wiki/Stemming) and
+> ["tokenization" (Wikipedia)](https://en.wikipedia.org/wiki/Lexical_analysis#Tokenization)
+> will be useful for understanding this section.
+
+Text analysis or tokenization is the broad process Elasticsearch follows to
+derive the index tokens for a given text document. A significant aspect of this
+for Openverse (and many applications) is
+[stemming](https://www.elastic.co/guide/en/elasticsearch/reference/7.12/stemming.html).
+When Elasticsearch performs a full text search, it is searching the derived
+tokens, for which it has created quickly searchable index, rather than the
+original text itself. This means that the text analysis configuration applied
+has a significant impact on the way documents are searched and how relevance is
+calculated.
+
+Openverse only applies our custom text analysis configuration to the "title",
+"description", and "tags" fields. All other fields use the default analyser.
+
+Openverse is currently configured exclusively with English language text
+analysis. This means Openverse does not properly index documents in any language
+other than English. This is a known issue that we hope to be able to address
+soon via Elasticsearch's internationalisation tools.
+
+We primarily use the default English language stemming settings aside from
+[minor changes (GitHub)](https://github.com/cc-archive/cccatalog-api/issues/574#issue-668091876)
+present in the `es_mapping.py` configuration to address a specific issue with
+the "anim" stem. We use the
+[Snowball English stemmer](https://snowballstem.org/algorithms/porter/stemmer.html)
+and
+[Lucene's possessive English stemmer](https://lucene.apache.org/core/8_8_0/analyzers-common/org/apache/lucene/analysis/en/EnglishPossessiveFilter.html).
+
+The rationale for this specific stemmer configuration is not documented aside
+from the "anim" stem issue linked above.
+
+Openverse also applies the `lowercase` text analysis filter making our indexes
+case-insensitive.
+
+## Search execution
+
+There are essentially 2 broad types of "full-text" searches available via the
+Openverse API:
+
+1. General "query" searching, which applies a simple text query search against
+   several, non-configurable fields.
+2. Individual field querying, which enables searching specific fields with
+   independent query terms.
+
+Both of these search types also allow further keyword filtering along the
```

Comment:
```suggestion
Both of these search types also allow further filtering of the results by the
```

Reply:
Added a suggestion because I feel like "keyword filtering" is not a very clear term (for someone without the knowledge of ES-specific terms)

Reply:
I added a term for "Keyword field" in the terms list. Keywords are a big part of how Openverse searches documents, so it seems like something worth understanding when reading this document. Let me know if I should add additional external resources for the concept of a "keyword" field. I did remove the word "along" though to simplify the text, thank you for the suggestion.